server/services/withings/api/withings.controller.js

```diff
@@ -0,0 +1,115 @@
+const asyncMiddleware = require('../../../api/middlewares/asyncMiddleware');
+
+module.exports = function WithingsController(gladys, withingsHandler) {
+  /**
+   * @description Init gladys devices with withings devices.
+   * @api {post} /api/v1/service/withings/init_devices Init gladys devices with withings devices.
+   * @apiName init
+   * @apiGroup Withings
+   */
+  async function init(req, res) {
+    const resultInit = await withingsHandler.initDevices(req.user.id);
+    res.json({
+      withingsDevices: resultInit,
+    });
+  }
+
+  /**
+   * @description Delete client id, secret client of withings oauth2 api and al withings devices.
+   * @api {get} /api/v1/service/withings/reset Delete clientId and secret of withings oauth2 api.
+   * @apiName reset
+   * @apiGroup Withings
+   */
+  async function reset(req, res) {
+    await withingsHandler.deleteVar(req.user.id);
+    await withingsHandler.deleteDevices();
+    res.json({
+      success: true,
+    });
+  }
+
+  /**
+   * @description Build an authorization uri (to get authorizationcode).
+   * @api {post} /api/v1/service/withings/oauth2/client/authorization-uri Build an authorization uri
+   * @apiName BuildAuthorizationUri
+   * @apiGroup OAuth2
+   */
+  async function buildAuthorizationUri(req, res) {
+    if (req.body && req.body.service_id && req.body.integration_name && req.headers.referer) {
+      const authorizationUriResult = await withingsHandler.oauth2Client.buildAuthorizationUri(
+        req.body.service_id,
+        req.user.id,
+        req.body.integration_name,
+        req.headers.referer,
+      );
+      res.json({
+        authorizationUri: authorizationUriResult,
+      });
+    } else {
+      res.status(400);
```

Comment:
So just 400, no error message, nothing? Also, we have native Error in Gladys, please use them (utils/httpErrors.js)

Reply:
I change by  throw new Error400(ERROR_MESSAGES.MISSING_MANDATORY_PARAMETER);